src/Analyzer.Utilities/Extensions/INamedTypeSymbolExtensions.cs

```diff
@@ -124,7 +124,113 @@ public static bool HasFinalizer(this INamedTypeSymbol symbol)
             return symbol.GetMembers()
                 .Where(m => m.Kind == SymbolKind.Method)
                 .Cast<IMethodSymbol>()
-                .Any(IMethodSymbolExtensions.IsFinalizer);
+                .Any(m => m.IsFinalizer());
+        }
+
+        public static bool HasAttribute(this INamedTypeSymbol symbol, INamedTypeSymbol attribute)
```

Comment:
Do we want to also include attributes on the base type/interfaces? Probably add a flag to control that? Regardless, can we add a comment so the consumer of the API is aware of the functionality?